### ISSUE TITLE
Necessity of com.google.android.gms.permission.AD_ID

### ISSUE THREAD

**Issue by jaakko-tiqets** [adjust/android_sdk#536]
In `README.md` you list adding `com.google.android.gms.permission.AD_ID` permission as one of the setup steps ([the section in README](https://github.com/adjust/android_sdk#add-permission-to-gather-google-advertising-id)).

Does Adjust SDK use the advertising ID for something other than providing a way to obtain the ID through `Adjust.getGoogleAdId()` ([as mentioned here in your README](https://github.com/adjust/android_sdk#google-play-services-advertising-identifier))?

In other words, if we do not use the `Adjust.getGoogleAdId()` method in our app, do we still need the `com.google.android.gms.permission.AD_ID` in order for Adjust to function properly?

Especially if the permission is not absolutely required, the actual usecase of the advertisement ID could maybe be made a bit clearer in `README.md`. 

Thanks 🙂

**Comment by uerceg** [comment]
Hi @jaakko-tiqets 

Yes, we do use it for something other than that getter you mentioned. Adjust is mobile analytics and attribution company and in order for us to be able to determine attribution (origin of where install comes from) for the users, access to Google Advertising ID is absolutely fundamental part of that flow. We are not mentioning this in our README because we just assume that Adjust SDK is not something people would randomly discover in GitHub and use, but rather use it for the above mentioned purpose where we again assume that people should be aware of Google Advertising ID purpose. :) But I'd say that this knowledge is more "natural" for UA type of people in companies to be aware of, definitely more than software engineers.

So to answer to your question - whether you need to add `com.google.android.gms.permission.AD_ID` permission to your app or not really depends on what are you using Adjust platform for. If you are running user acquisition campaigns (and if you are actually interested in attribution at all) for your app(s) in Play Store, chances are pretty high that you do need access to Google Advertising ID. This is at the end of the day something you would need to check with your stakeholders who are working with Adjust.

At the end of the day, not adding this permission is not something which will prevent SDK to perform its basic functionality. Everything will work as it normally would from SDK point of view, it's just that all the things we send to our backend would be sent without Google Advertising ID parameter. But like mentioned above, what lack of this parameter might cause is that some use cases you are trying to cover with usage of Adjust platform might not work.

In case you have any other questions, feel free to ping.

**Comment by jaakko-tiqets** [comment]
Tracking the sources of install is indeed what we mainly use Adjust for - so I take it we do want to have the permission in.

Thank you for a thorough and fast reply 🙏. I'll close this issue.

**Comment by lwld** [comment]
Can I use this ticket to ask why is the `com.google.android.gms.permission.AD_ID` now "enforced" in the SDK's manifest (dc257c3f169c74f513b9eced5211acdeee4ab112) in the newest release (https://github.com/adjust/android_sdk/releases/tag/v4.31.1)?

What has changed to the explanation of @uerceg above?

**Comment by uerceg** [comment]
Hi @lwld,

Yes, you can absolutely use this ticket to ask this question and great question, indeed.

Requests for this feature (SDK auto-adding this particular permission in lights of Android platform enforcing it in post API 33 world) became really numerous and coming in from a lot of our clients. That's why we proceeded with this shift where we we are adding this permission by default and in our [docs](https://help.adjust.com/en/article/get-started-android-sdk#add-permissions) we are advising people how can they remove it in case they don't want it to be added by default.

In case you have any other questions, feel free to ask.

**Comment by lwld** [comment]
Thanks!
